### PR TITLE
Plugins configuration docs - format parameters as a table

### LINKED ISSUE #280: Plugins configuration docs: format parameters as a table
Per @ccapell suggestion it would be much cleaner and user-friendly if we'd format the https://bitovi.github.io/bitops/configuration-base/ Configuration documentation for plugins, tools, and Cloud providers with a table instead of using a list that takes long scrolling and uses lots of space.

image

## TODO
- [ ] Update the subpages in the https://bitovi.github.io/bitops/configuration-base/
- [ ] Update every plugin README in the https://github.com/bitops-plugins

### PR DESCRIPTION
**Pull request by ccapell**
# Description

Documentation has been formatted to be a table instead of a list

Fixes #280

## Type of change

Please delete options that are not relevant.

- [x] Documentation update

### CODE REVIEW

File: docs/tool-configuration/configuration-terraform.md [hunk 1]
```diff
@@ -34,65 +34,20 @@ Run BitOps with the environmental variable `TERRAFORM_APPLY` set to `true` or se
...
-### workspace
-* **BitOps Property:** `workspace`
-* **Environment Variable:** `BITOPS_TERRAFORM_WORKSPACE`
-* **default:** `""`
-* **Required:** `false`
-* **Description:** Will select a terraform workspace using `terraform workspace new $TERRAFORM_WORKSPACE || terraform workspace select $TERRAFORM_WORKSPACE` prior to running other terraform commands.
+| Property     | Environment Variable       | Description                                                  | Default | Required |
+| ------------ | -------------------------- | ------------------------------------------------------------ | ------- | -------- |
+| stack-action | BITOPS_TERRAFORM_COMMAND   | Controls what terraform command to run. e.g. `apply`, `destroy`, etc. | `plan`  | No       |
+| version      | BITOPS_TERRAFORM_VERSION   | Allows customization of which version of terraform to run. **NOTE:** `This feature is currently not supported.` | `1.2.2` | No       |
```
Comment (armab) [inline docs/tool-configuration/configuration-terraform.md:49]:
```suggestion
```

Remove.
Something that's not present (non-implemented feature) shouldn't be part of the Documentation.

This was already a source of confusion in the https://github.com/bitovi/bitops/issues/307

File: docs/tool-configuration/configuration-cloudformation.md [hunk 2]
```diff
@@ -20,113 +20,41 @@ cloudformation:
...
-* **Environment Variable:** `CFN_STACK_ACTION`
-* **default:** `deploy`
-
-Controls what cloudformation action to apply on the stack
-
--------------------
+| Property         | Environmental Variable | Description                                               | Default  | Required |
+| ---------------- | ---------------------- | --------------------------------------------------------- | -------- | -------- |
+| validate-cfn     | FN_TEMPLATE_VALIDATION | Calls `aws cloudformation validate-template`              | `true`   |          |
+| cfn-stack-action | CFN_STACK_ACTION       | Controls what CloudFormation action to apply on the stack | `deploy` |          |
```
Comment (armab) [inline docs/tool-configuration/configuration-cloudformation.md:26]: Something is broken here with formatting
image

File: docs/cloud-configuration/configuration-aws.md [hunk 3]
```diff
@@ -1,45 +1,15 @@
...
 
--------------------
-### aws_access_key_id
-* **BitOps Property:** [TODO](https://github.com/bitovi/bitops/issues/15)
-* **Environment Variable:** `AWS_ACCESS_KEY_ID`
-* **default:** `""`
-* **required:** yes
+| Item                  | BitOps Property                                    | Environmental Variable | Description                                                  | Default | Required |
+| --------------------- | -------------------------------------------------- | ---------------------- | ------------------------------------------------------------ | ------- | -------- |
+| aws_access_key_id     | [TODO](https://github.com/bitovi/bitops/issues/15) | AWS_ACCESS_KEY_ID      | Specifies an AWS access key associated with an IAM user or role. See [AWS official documentation](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) | `null`  | Yes      |
```
Comment (armab) [inline docs/cloud-configuration/configuration-aws.md:11]: @mickmcgrath13 @PhillypHenning Do we know what these `TODO` links really mean?
Any history context and can we remove those?

There is no `BitOps Property` columns in the other table.

File: docs/tool-configuration/configuration-helm.md [hunk 4]
```diff
@@ -29,145 +29,44 @@ helm:
...
-simulate an install
+| Property  | Environment Variable | Description                                                  | Default | Required |
+| --------- | -------------------- | ------------------------------------------------------------ | ------- | -------- |
+| namespace | NAMESPACE            | Namespace scope for this project                             | `null`  | Yes      |
+| timeout   | TIMEOUT              | Time to wait for any individual Kubernetes operation (like Jobs for hooks) | `500s`  |          |
+| set       | HELM_SET_FLAG        | List of "key=value" strings to pass in to `helm` via `--set` | `{}`    |          |
+| debug     | HELM_DEBUG           | Enable verbose helm output                                   | `null`  |          |
+| atomic    |                      | If set, the installation process deletes the installation on failure | `null`  |          |
+| force     |                      | Sets helm's `--force` flag                                   | `null`  |          |
+| dry-run   |                      | Simulate an install                                          | `null`  |          |
```
Comment (armab) [inline docs/tool-configuration/configuration-helm.md:40]: Looking at the tables, does it make sense to move the `Default` column right before `Description`?
I mean, in the context of Key=Value this is easier to understand.

image

@ccapell any reasons behind composing it that way?
Curious what other folks think @dylan-bitovi @LeoDiazL @mickmcgrath13 @PhillypHenning

**Review by armab — CHANGES_REQUESTED** [review]
Thanks a lot for the changes!
I like the diff here: `+108 −549`, - fewer lines and so much improved visual clarity by formatting it as a table 👍 

I left several comments to address and clarify before we merge it.

File: docs/tool-configuration/configuration-helm.md (hunk 4 above)
Comment (LeoDiazL) [reply docs/tool-configuration/configuration-helm.md:40]: Love the idea of the table. Gives you a cleaner view.

Regarding default position, it does makes sense next to the environment variable, as long as it adds something. If it's going to be mainly null or a default you won't change, I'll leave it there.

File: docs/tool-configuration/configuration-terraform.md (hunk 1 above)
Comment (ccapell) [reply docs/tool-configuration/configuration-terraform.md]: Removed

File: docs/tool-configuration/configuration-cloudformation.md (hunk 2 above)
Comment (ccapell) [reply docs/tool-configuration/configuration-cloudformation.md:26]: Fixed

File: docs/cloud-configuration/configuration-aws.md (hunk 3 above)
Comment (ccapell) [reply docs/cloud-configuration/configuration-aws.md:11]: Those TODO's link to issue #15 and were in the original documentation

File: docs/tool-configuration/configuration-helm.md (hunk 4 above)
Comment (ccapell) [reply docs/tool-configuration/configuration-helm.md:40]: @armab I was just adding the parameters in the order that they appeared.  Most of the tables have very few if any default values.  For consistency stake, all the tables have the same order.

File: docs/cloud-configuration/configuration-aws.md (hunk 3 above)
Comment (ccapell) [reply docs/cloud-configuration/configuration-aws.md:11]: So this one has an 'Extra' column, named Item.  The property column is also called 'BitOps Property' in this table.   As I formatted the information into a table, I wanted to ensure that I kept the same naming conventions as there were original.

**Review by ccapell — COMMENTED** [review]
Changes made

**Comment by PhillypHenning** [comment]
Screen Shot 2022-10-31 at 11 26 33 AM

**Comment by PhillypHenning** [comment]
Tables look great @ccapell 

Screen Shot 2022-11-02 at 3 06 15 PM

**Review by PhillypHenning — APPROVED** [review]

**Review by PhillypHenning — CHANGES_REQUESTED** [review]
Hey @ccapell could you please table-ize the `additional information` sections for all the plugins 👍

**Review by PhillypHenning — APPROVED** [review]

**Comment by PhillypHenning** [comment]
Pigging back on this PR to apply the following [doc update](https://github.com/bitovi/bitops/issues/343)

**Pull request merged**